lightningd/opening_control.c

```diff
@@ -1202,6 +1203,184 @@ static struct command_result *json_fundchannel_start(struct command *cmd,
 
+static struct channel *stub_chan(struct command *cmd,
+								u64 id,
+								struct node_id nodeid, 
+								struct channel_id cid,
+								struct bitcoin_outpoint funding,
+								struct wireaddr_internal addr){
+	struct peer *peer;
+	peer = new_peer(cmd->ld,
+						0,
+						&nodeid,
+						&addr,
+						false
+						);
+	
+	struct lightningd *ld = cmd->ld;
+	
+
+	struct channel_config our_config;
+	memset(&our_config, 1, sizeof(struct channel_config));
+	our_config.id = 0;
+	
+	u32 feerate;
+	/* FIXME: What should be the appropriate value here? */
+	feerate = 192838; 
+
+	struct bitcoin_signature sig;
+	sig.sighash_type = SIGHASH_ALL;
+
+	struct pubkey pk;
+	struct basepoints basepoints;
+	pubkey_from_der(tal_hexdata(ld->wallet, 
+					type_to_string(tmpctx, struct node_id, &nodeid), 66),
+					33,
+					&pk);
+
+	struct pubkey localFundingPubkey;
+	get_channel_basepoints(ld, 
+						   &nodeid, 
+						   id, 
+						   &basepoints, 
+						   &localFundingPubkey);
+
+	struct channel_info *channel_info = tal(ld->wallet, 
+											struct channel_info);
+	memset(channel_info, 3, sizeof(*channel_info));
+	channel_info->their_config.id = 0;
+	channel_info->theirbase = basepoints;
+	channel_info->remote_fundingkey = pk;
+	channel_info->remote_per_commit = pk;
+	channel_info->old_remote_per_commit = pk;
+
+	u32 blockht = 100;
+	
+	struct short_channel_id *scid = tal(cmd, struct short_channel_id);
+	
+	/*To indicate this is an stub channel we keep it's scid to 1x1x1.*/
+	if(!short_channel_id_from_str("1x1x1", 6, scid)){
+		return NULL;
+	}
+	
+	struct channel *channel; //Channel Shell with Dummy data(mostly)
+
+	channel = new_channel(peer, id,
+			      NULL, /* No shachain yet */
+			      CHANNELD_NORMAL,
+			      LOCAL,
+			      peer->ld->log,
+			      "billboard",
+			      8, &our_config,
+			      101,
+			      1, 1, 1,
+			      &funding,
+			      AMOUNT_SAT(500000000), // FIXME: Is this correct? 
+				  /*Need to keep this very high from trimming the fees 
+				  	in onchaind.c(main).*/
+			      AMOUNT_MSAT(0),
+			      AMOUNT_SAT(0),
+			      true, /* !remote_funding_locked */
+			      scid, /* no scid yet */
+			      &cid,
+			      /* The three arguments below are msatoshi_to_us,
+			       * msatoshi_to_us_min, and msatoshi_to_us_max.
+			       * Because, this is a newly-funded channel,
+			       * all three are same value. */
+			      AMOUNT_MSAT(0),
+			      AMOUNT_MSAT(0), /* msat_to_us_min */
+			      AMOUNT_MSAT(0), /* msat_to_us_max */
+			      NULL,
+			      &sig,
+			      NULL, /* No HTLC sigs yet */
+			      channel_info,
+			      new_fee_states(ld->wallet, LOCAL, &feerate),
+			      NULL, /* No shutdown_scriptpubkey[REMOTE] yet */
+			      NULL,
+			      1, false,
+			      NULL, /* No commit sent yet */
+			      /* If we're fundee, could be a little before this
+			       * in theory, but it's only used for timing out. */
+			      get_network_blockheight(ld->topology),
+			      100, 10000,
```

Comment:
onchaind needs these to be correct, (min and max possible feerate) which is a problem!  It iterates through this range to look for where the signatures match (so it doesn't have to remember each fee value for each tx).

min is easy: FEERATE_FLOOR.  max is harder: if we set it to -1ULL (max u64), onchaind will spin forever: see line 204!  But we can assume it's less than the funding, if we store that in the scb, and we have a MINIMUM_TX_WEIGHT constant, so we can actually set it to:

```
funding.satoshis / MINIMUM_TX_WEIGHT * 1000 /* Raw: convert to feerate */
```
(Note that you're not normally supposed to access fields on amount_sat, so the comment suppresses the warning!).